src/open_inwoner/react/modules/Sidenav/Sidenav.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+import { SideNavigation } from '@gemeente-denhaag/side-navigation'
+
+// Material Icons component
+const MaterialIcon: React.FC<{ name: string }> = ({ name }) => (
+  <span className="material-icons-outlined" aria-hidden="true">
+    {name}
+  </span>
+)
+
+const Sidenav: React.FC = () => {
+  // Get menu data from Django
+  const getMenuData = () => {
+    const scriptElement = document.getElementById('sidenav-menu-data')
+    if (scriptElement?.textContent) {
+      try {
+        const data = JSON.parse(scriptElement.textContent)
+        console.log('Menu data loaded:', data) // Debug log
+        return data
+      } catch (e) {
+        console.error('Failed to parse menu data:', e)
+      }
+    }
+    return []
+  }
+
+  const menuData = getMenuData()
```

Comment:
If you are going to implement the generic json change the fn to this to make sure menuData has the correct type.
```suggestion
  interface MenuItem {
    href: string
    label: string
    icon?: string
    counter?: number
    current?: boolean
  }

  const menuData = getJsonScriptData<MenuItem>()
```